### DISCUSSION TITLE
What's the difference between a Turbo app and a Turbo package?

### DISCUSSION THREAD

**Discussion by wujekbogdan** [vercel/turborepo#9420] [Help] [answered]
### Summary

I understand that the difference between an app and a package is that an app is a deployable unit (something we can wrap in a Docker container and deploy), while a package is just a library - a dependency shared between apps and other packages.

Up until now, I thought that placing one under an `apps` directory and the other in the `packages` directory was just a convention. However, for Turbo, there's no inherent distinction between these two, as they are all just npm/pnpm/yarn workspaces.

But I found two places in the documentation that indicate Turbo _might_ apply some special treatment depending on whether it's an app or a regular package:

- The [Package Types documentation](https://turbo.build/repo/docs/core-concepts/package-types#application-packages) mentions this distinction.
- [The `turbo gen` command has a `--type` argument](https://turbo.build/repo/docs/reference/generate#--type-apppackage) that accepts either `app` or `package` as a value.

Does this mean that Turbo makes some distinction between apps and packages, or is it just the documentation that's misleading and the `--type` argument accepts any value that is simply a workspace name?

**Comment by anthonyshew** [comment, marked as answer]
You're correct that this is only a convention. Turborepo carries no distinction on these directories.

`turbo gen` does leverage those conventions. However, it's more likely that we should find a way for `turbo gen` to not have this behavior.
